joomla/joomla-cms
# [4.0] Media field, Media Manager fixes

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/29147 .

### Summary of Changes
#### Media Manager: 
- remove the logic for dispatching event (used in the media field) from the item component to the store
- Add dispatch event for all the single select item operations
- Add dispatch event for the unselect item operation

#### Media field
- Remove couple of Joomla scoped variables/functions (not needed)
- Tweak the fetchImageDetails to reject immediately if there is no selected image in media manager
- Proper use of the promise when calling the fetchImageDetails fn

### Testing Instructions
Case 1: Normal operation
Create a new article and insert an image to the intro image.
Reset the field by pressing the remove button

Case 2: Selection problem
Create an article and go to the intro image opening the modal for the Media manager. Select an image by clicking on it. Now click anywhere in the white space below the list of images. The image should be unselected! Clicking the select button at the end of the modal should:
- Not insert anything in the field (this was previously returning the last image selected)
- Close the modal (same behaviour as 3.x)

Case 3: Race Condition
I can safely replicate this by adding a breakpoint and using debugger in the PHP world.
Basically what happens is that selecting an image is not synchronous so if you're too fast to select an image and then click on the select button on the bottom of the modal you will not get the selected image path in the input field. Probably can be reproduced with the dropbox plugin(?)


### Expected result
All cases should work


### Actual result



### Documentation Changes Required

Nope
